packages/marshal/marshal.js

```diff
@@ -564,3 +641,83 @@ export function makeMarshal(
     unserialize,
   });
 }
+
+/**
+ * Create and register a Presence.  After this, getInterfaceOf(presence)
+ * returns iface.
+ *
+ * // https://github.com/Agoric/agoric-sdk/issues/804
+ *
+ * @param {string} [iface='Presence'] The interface specification for the presence
+ * @param {object} [props={}] Own-properties are copied to the presence
+ * @param {object} [presence={}] The object used as the presence
+ * @returns {object} presence, modified for debuggability
+ */
+function Presence(iface = 'Presence', props = {}, presence = {}) {
+  iface = pureCopy(iface);
+  const ifaceType = typeof iface;
+
+  // Find the alleged name.
+  if (ifaceType !== 'string') {
+    throw Error(`Interface must be a string, not ${ifaceType}; unimplemented`);
+  }
+
+  // Fail fast: check that the unmodified object is able to become a Presence.
+  assertCanBePresence(presence);
+
+  // Ensure that the presence isn't already registered.
+  if (presenceToInterface.has(presence)) {
+    throw Error(`Presence ${presence} is already mapped to an interface`);
+  }
+
+  // TODO: Get the allegedName a different way.
+  const allegedName = iface;
+
+  // A prototype for debuggability.
+  const oldPresenceProto = Object.getPrototypeOf(presence);
+
+  // Fail fast: check the old proto against our rules.
+  mustPassByPresence(harden(Object.create(oldPresenceProto)));
```

Comment:
This rejects if `presence` is not frozen. Where to you earlier reject if presence is frozen? Where do you freeze or harden it?

Reply:
I've made this clearer.